airflow/contrib/hooks/aws_glue_job_hook.py

```diff
@@ -0,0 +1,259 @@
+        except Exception as general_error:
+            raise AirflowException(
+                'Failed to run aws glue job, error: {error}'.format(
+                    error=str(general_error)
+                )
+            )
+
+    def _job_completion(self, job_name=None, run_id=None):
+        """
+        :param job_name:
+        :param run_id:
+        :return:
+        """
+        glue_client = self.get_conn()
+        job_status = glue_client.get_job_run(
+            JobName=job_name,
+            RunId=run_id,
+            PredecessorsIncluded=True
+        )
+        job_run_state = job_status['JobRun']['JobRunState']
+        failed = job_run_state == 'FAILED'
+        stopped = job_run_state == 'STOPPED'
+        completed = job_run_state == 'SUCCEEDED'
+
+        if failed or stopped or completed:
+            self.log.info("Exiting Job {} Run State: {}".format(run_id, job_run_state))
+            return job_run_state
+        else:
+            self.log.info("Polling for AWS Glue job {} status".format(job_name))
+            self.log.info("Current AWS Glue Job State: {}".format(job_run_state))
+            return self._job_completion(job_name, run_id)
+
+    def _get_or_create_glue_job(self):
+        glue_client = self.get_conn()
+        try:
+            self.log.info("Now creating and running AWS Glue Job")
+            s3_log_path = "s3://{bucket_name}/{logs_path}{job_name}"\
+                .format(bucket_name=self.s3_bucket,
+                        logs_path=self.S3_GLUE_LOGS,
+                        job_name=self.job_name)
+
+            execution_role = self._create_job_execution_role()
+            script_location = self._check_script_location()
+            create_job_response = glue_client.create_job(
+                Name=self.job_name,
+                Description=self.desc,
+                LogUri=s3_log_path,
+                Role=execution_role['Role']['RoleName'],
+                ExecutionProperty={"MaxConcurrentRuns": self.concurrent_run_limit},
+                Command={"Name": "glueetl", "ScriptLocation": script_location},
+                MaxRetries=self.retry_limit,
+                AllocatedCapacity=self.num_of_dpus
+            )
+            return create_job_response['Name']
+        except Exception as general_error:
+            raise AirflowException(
+                'Failed to create aws glue job, error: {error}'.format(
+                    error=str(general_error)
+                )
+            )
+
+    def _check_script_location(self):
+        """
+        :return: S3 Script location path
+        """
+        if self.script_location[:5] == self.S3_PROTOCOL:
+            return self.script_location
+        elif os.path.isfile(self.script_location):
+            s3 = self.get_resource_type('s3', self.region_name)
+            script_name = os.path.basename(self.script_location)
+            s3.meta.client.upload_file(self.script_location,
```

Comment:
maybe add the job_name here as well to make the location unique per job

Reply:
I see you modified the `s3_script_path` variable to include the job_name, but you're still uploading to a location without the job_name